lib/vector-core/src/event/vrl_target.rs

```diff
@@ -20,18 +20,18 @@ const MAX_METRIC_PATH_DEPTH: usize = 3;
 pub enum VrlTarget {
     // `LogEvent` is essentially just a destructured `event::LogEvent`, but without the semantics
     // that `fields` must always be a `Map` variant.
-    LogEvent(Value, EventMetadata),
+    LogEvent(Rc<RefCell<vrl_core::Value>>, EventMetadata),
```

Comment:
The `LogEvent` is now a `vrl::Value` rather than a Vector `Value`. 

This means we no longer have to convert between Vector and Vrl every time a field is read and written to. The conversion occurs once at the start of the process, and then converted back to a Vector `Value` at the end.